designs/FloatingStableAndPrerelease.md

```diff
@@ -0,0 +1,135 @@
+
+# Support pre-release packages with floating versions
+
+* Status: **In Review**
+* Author(s): [Nikolche Kolev](https://github.com/nkolev92)
+* Issue: [912](https://github.com/NuGet/Home/issues/912) Support pre-release packages with floating versions
+* Type: Feature
+
+## Problem Background
+
+Consider a PackageReference as such:
+
+```xml
+<ItemGroup>
+    <PackageReference Include="NuGet.Packaging" Version="5.*" />
+</ItemGroup>
+```
+
+Right now this will only include stable package versions.
+
+| Floating Version | Available Versions | Best matching version |
+| -----------------|--------------------|-----------------------|
+| 5.* | 5.0.0 <br> 5.1.0 <br> 5.2.0 <br> 5.3.0 <br> 5.4.0 <br> 5.5.0-preview.1 <br> 5.5.0-preview.2 | 5.4.0 |
+
+There is no way to float to both stable and prerelease versions at the same time.
+
+## Who are the customers
+
+PackageReference customers. A popular customer issue with > 80 comments & 38+ up-votes.
+
+## Requirements
+
+From the issue:
+
+* I would like to use floating versions that only resolve to stable versions of the package (Current behavior)
+* I would like to use floating versions that can resolve to the latest version even if the latest happens to be a pre-release version.
+* I would like to additionally define whether I want to resolve to rc, beta, alpha or all pre-releases.
+
+## Goals
+
+* Define a versioning strategy that allows floating to both stable and prerelease versions simultaneously. Floating only prerelease and only stable versions is possible today with `1.0.0-*` and `1.*` respectively.
+
+## Non-Goals
+
+* Support for more than 1 prerelease label in floating versions. This is out of scope, because there is no way today with the prerelease only floating versions. As such it can be treated as a separate feature.

```

Comment:
I don't understand this statement. My understanding is that pre-release labels are sorted, i.e. `beta` would be considered newer than `alpha`. In that sense it supports multiple labels? Or are you saying that consumers won't be able to say "resolve latest in the `beta` band", as opposed to "latest version across all labels"?

Reply:
When you say `1.1.0-alpha.*` it means only the prereleases with an alpha label will be preferred. 
See [example](https://nugettoolsdev.azurewebsites.net/5.5.0-floating.7611/find-best-version-match?versionRange=1.1.0-alpha.*&versions=1.1.0-alpha.3%0D%0A1.1.0-beta.1%0D%0A1.1.0). 

If you specify `1.1.0-*` then all `1.0.0-` prerelease versions are in the preferred range. 

See similar [example](https://nugettoolsdev.azurewebsites.net/5.5.0-floating.7611/find-best-version-match?versionRange=1.1.0-*&versions=1.1.0-alpha.3%0D%0A1.1.0-beta.1%0D%0A1.1.0) how the priority changes. 

So customers can say: 

* Resolve latest beta
* Resolve latest version across all labels

but can't say

* Resolve latest alpha or beta. 

Does that make sense? 
I think it's worth adding an explanation but I want to confirm it's understandable.